plugins/statistics.koplugin/main.lua

```diff
@@ -866,6 +866,83 @@ function ReaderStatistics:getCurrentStat(id_book)
     }
 end
 
+function ReaderStatistics:getBookStat(id_book)
+    if id_book == nil then
+        return
+    end
+    local conn = SQ3.open(db_location)
+    local sql_stmt = [[
+        SELECT title, authors, notes, highlights, pages, last_open
+        FROM book
+        WHERE id = '%s'
+    ]]
+    local title, authors, notes, highlights, pages, last_open = conn:rowexec(string.format(sql_stmt, id_book))
+
+    sql_stmt = [[
+        SELECT count(*)
+        FROM   (
+                    SELECT strftime('%%Y-%%m-%%d', start_time, 'unixepoch', 'localtime') AS dates
+                    FROM   page_stat
+                    WHERE  id_book = '%s'
+                    GROUP  BY dates
+               )
+    ]]
+    local total_days = conn:rowexec(string.format(sql_stmt, id_book))
+
+    sql_stmt = [[
+        SELECT sum(period),
+               count(DISTINCT page)
+        FROM   page_stat
+        WHERE  id_book = '%s'
+    ]]
+    local total_time_book, total_read_pages = conn:rowexec(string.format(sql_stmt, id_book))
+
+    sql_stmt = [[
+        SELECT min(start_time)
+        FROM   page_stat
+        WHERE  id_book = '%s'
+    ]]
+    local first_open = conn:rowexec(string.format(sql_stmt, id_book))
+    conn:close()
+
+    if total_time_book == nil then
+        total_time_book = 0
+    end
+    if total_read_pages == nil then
+        total_read_pages = 0
+    end
+    total_time_book = tonumber(total_time_book)
+    total_read_pages = tonumber(total_read_pages)
+    pages = tonumber(pages)
+    if pages == nil or pages == 0 then
+        pages = 1
+    end
+    local avg_time_per_page = total_time_book / total_read_pages
+    return {
+        { _("Title"), title},
+        { _("Authors"), authors},
+        { _("First book open"), os.date("%Y-%m-%d (%H:%M)", tonumber(first_open))},
```

Comment:
How about "first opened" and "last opened"? (Or possibly "Book first opened".)

Reply:
"first opened" sounds better :)